File: selenium/src/test/java/com/ytfc/blog/pages/user/ForgotPasswordPage.java
package com.ytfc.blog.pages.user;

import com.ytfc.blog.utility.Helper;
import com.ytfc.blog.forms.ForgotPasswordForm;
import com.ytfc.blog.pages.PageObject;

import org.openqa.selenium.WebDriver;

public class ForgotPasswordPage extends PageObject {
    private ForgotPasswordForm form;

    public ForgotPasswordForm getForm() {
        return this.form;
    }

    public ForgotPasswordPage(WebDriver driver) {
        this.driver = driver;
        this.form   = new ForgotPasswordForm(driver);
    }

    public ForgotPasswordPage get() {
        Helper.get(driver, "user/forgot_password");

        assertPageTitle("Forgot password", "Page has correct title");
        form.assertFieldTypes("Field has correct type");

        return this; // Chainable
    }

    public void submit(String reqEmail) {
        get();
        form.getField("email").setValue(reqEmail);
        form.getButton().click();
    }

}

File: cypress/pages/forgot-password-page.js
import { Public             } from '../pages/public'
import { ForgotPasswordForm } from '../forms/forgot-password-form'
import { LoginPage          } from '../pages/login-page'

export class ForgotPasswordPage extends Public {
    constructor() {
        super()
        this._form = new ForgotPasswordForm()
    }

    visit() {
        new LoginPage().clickForgotPasswordLink()
        return this;
    }


    forgotPassword(email) {
        this.getForm().enter(new Map([['email', email]]))

        return this
    }

}
